tasks/homework.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"math/rand/v2"
+)
+
+type ZooKeeper struct {
+	Name string
+}
+
+func (keper *ZooKeeper) SendToCage(animal *Animal, cage *Cage) {
+	cage.Animal = animal
+	animal.IsFree = false
+}
+
+type Animal struct {
+	Name   AnimalType
+	IsFree bool
+}
+
+func (a *Animal) makeFree() {
+	a.IsFree = true
+}
+
+type AnimalType string
+
+const (
+	Lion   AnimalType = "Lion"
+	Tiger  AnimalType = "Tiger"
+	Duck   AnimalType = "Duck"
+	Monkey AnimalType = "Monkey"
+)
+
+type Cage struct {
+	Animal *Animal
+}
+
+type Zoo struct {
+	City  string
+	Cages []Cage
+}
+
+func NewZoo(city string) *Zoo {
+	return &Zoo{City: city}
+}
+
+func (cage *Cage) open() {
+	cage.Animal = nil
+}
+
+func main() {
+	animalsNames := [4]AnimalType{Lion, Tiger, Duck, Monkey}
+	animalsAmount := rand.IntN(10)
+
+	var animals []Animal
+
+	zoo := NewZoo("Kyiv")
+	fmt.Printf("There are %d animals in the %s zoo\n", animalsAmount+1, zoo.City)
+
+	zooKeper := ZooKeeper{"Bob"}
+
+	// Create animals and cages
+	for i := 0; i <= animalsAmount; i++ {
+		name := animalsNames[rand.IntN(4)]
+		animal := Animal{name, false}
+		animals = append(animals, animal)
+		fmt.Printf("%s in cage\n", animals[i].Name)
+		zoo.Cages = append(zoo.Cages, Cage{&animal})
+	}
+
+	// Opening cells randomly
+	for i, _ := range zoo.Cages {
+		isOpen := rand.IntN(2) == 0
+		if isOpen == true {
+			zoo.Cages[i].open()
+			animals[i].IsFree = true
```

Comment:
В кожної тварини є метод "втекти", може краще його використати?
```suggestion
			animals[i].makeFree()
```

Reply:
так, додав виклик